### ISSUE TITLE
[Bug] Spectral Arrow glowing effect not working

### ISSUE THREAD

**Issue by MrMazw** [sp614x/optifine#3662]
## Description of Issue
Please provide as much detail as possible.
When you either use Spectral Arrows or just try to give yourself the glowing effect it just doesn't work, the potion particles are there but the effect doesn't work

## Steps to Reproduce
1. What do you do... 
2. ...to find this bug?
Shoot something with a spectral arrow or use the /effect command to give glowing potion effect

## OptiFine Version
Self-explanatory. 
1.15.2 pre 13 but has also been in the past few versions to my knowledge

## Installation Method
How did you install OptiFine? Did you use the included standalone installer, or did you drop the JAR file into your "mods" folder?
Did the normal download from the download page 

## Log Files/Crash Reports
Any relevant log files. Drag and drop text files here to upload to GitHub directly, OR upload the complete contents to Pastebin. https://pastebin.com/

You can get logs from `.minecraft/logs`, and crash reports from `.minecraft/crash-reports`. 
Locating your `.minecraft` folder will depend entirely on your OS. See this page for details: https://minecraft.gamepedia.com/.minecraft

**DO NOT directly copy and paste log contents here on GitHub.**

## F3 Debug Screenshot
Minecraft's built-in debug screen provides *a lot* of useful information, and should be included even if the issue is not something that can be captured in a still image.

To open this menu, simply press `F3` on your keyboard while in-game. Please ensure the text is easily readable. Depending on the size/resolution of your monitor, you may have to adjust your GUI scale for all text to be completely visible on-screen.
![2020-04-02_18 07 56](https://user-images.githubusercontent.com/63076905/78304642-30aead00-750d-11ea-8d0c-b735ee7bf561.png)

## Prior Testing
If you've installed standalone:
- Does this happen in Vanilla without OptiFine?
No, the effect works normally in 1.15.2

If you've installed using Forge/Fabric:
- Does this happen without other mods installed? 
   - Only OptiFine and Forge
   - Only OptiFine, OptiFabric, and Fabric
- Does this happen without any mods at all?
   - Only Forge/Fabric by itself.
- Does this happen in OptiFine standalone?
- Does this happen in Vanilla without OptiFine?
- Have you tried using a binary search to find which mods cause this issue?
   - To find conflicting mods, split your mods into 2 groups (not including OptiFine/OptiFabric, obviously). Remove one group, and test in-game. Keep the group that has the problem, and repeat until no more mods can be removed without the issue disappearing.
   - Does the issue still happen with only the remaining mods installed?
      - Forge/Fabric and remaining mods, without OptiFine

## Additional Information
Here goes any final remarks and everything else that might not fit in the previous sections.

**Comment by Immabed** [comment]
Do you have "Performance > Fast Render" set to on? I can replicate this bug with Fast Render on, but glowing works with Fast Render turned off.

**Comment by MrMazw** [comment]
Yes turning that off fixed it for me thank you for that. Thought I'd
already messed around with all of the settings.

**Comment by Natalcia** [comment]
I'm not sure but glow effect is using shaders that fast render is breaking

**Comment by VolatileCable** [comment]
I have the same issue that glowing effect or nbt doesnt work whatever I try. Normally I play with internal shaders, fxaa and fast render off, but also tried other combos and nothing works

**Comment by MiKayule** [comment]
## Description
Confirmed that OptiFine 1.15.2 HD U G1 pre13 have this issue when Fast Render is enabled. I usually use OptiFine on Fabric clients with OptiFabric, and I have also tested OptiFine client, the issue exists on both clients.

## Crash Reports
(Some irrelevant information have deleted)
| Client | Crash Report |
| --- | --- |
| Fabric+OptiFabric+OptiFine | https://pastebin.com/6a0EAR3a |
| OptiFine only | https://pastebin.com/qMcK7SCd |

**Comment by runley** [comment]
i have the same issue rn even with fast render off

**Comment by naeqirelle2020** [comment]
still occurs in latest preview (pre-23)

**Comment by NickPlayzMC** [comment]
This bug also still occurs in pre-26.

**Comment by Cloud7050** [comment]
This happens with pre-30 1.15.2

**Comment by Felina-Lain** [comment]
Still happening with 1.16.3
(I tested with fast render off, but with a sildur shader)

**Comment by Cloud7050** [comment]
> Still happening with 1.16.3
> (I tested with fast render off, but with a sildur shader)

On 1.16.3 HD U G3 with fast render off and shaders off, things worked for me as expected. With internal shaders on instead, glowing did not work at all as was the case for me on 1.15.2 pre30. With Complementary shaders, the glowing did not work at all on 1.15 but on 1.16 it worked a little but was very buggy - a thick solid colour outline (unlike vanilla where the outline colour fades to black towards the edges) which included the player's floating IGN, with the entire player model fully visible through walls.

**Comment by Felina-Lain** [comment]
> > Still happening with 1.16.3
> > (I tested with fast render off, but with a sildur shader)
> 
> On 1.16.3 HD U G3 with fast render off and shaders off, things worked for me as expected. With internal shaders on instead, glowing did not work at all as was the case for me on 1.15.2 pre30. With Complementary shaders, the glowing did not work at all on 1.15 but on 1.16 it worked a little but was very buggy - a thick solid colour outline (unlike vanilla where the outline colour fades to black towards the edges) which included the player's floating IGN, with the entire player model fully visible through walls.

What complementary shaders did you use? I figure even a bugged looking effect would be better than none at all, so I'm interested.

**Comment by kwon0408** [comment]
1.16.3-HD-U-G4, SEUS Renewed Shader 1.0.1, no anti-aliasing, no fast render: Glowing effect doesn't show at all.

After changing the shader to Complementary Shaders (A BSL Edit) 3.07, the effect shows as I expected.

**Comment by kwon0408** [comment]
> What complementary shaders did you use? I figure even a bugged looking effect would be better than none at all, so I'm interested.

*Complementary Shaders* is the name of a certain shader.

**Comment by Felina-Lain** [comment]
> > What complementary shaders did you use? I figure even a bugged looking effect would be better than none at all, so I'm interested.
> 
> _Complementary Shaders_ is the name of a certain shader.

Oh, I thought it was an adjective XD
Alright thanks!

**Comment by LowJunnHoi** [comment]
Same thing happened to me. I'm using chocapic toaster.

**Comment by Ten2O** [comment]
Still not working, even with complementary shaders it does not show at all. It only shows with all shaders, including internal, turned off.

**Comment by Cloud7050** [comment]
> Still not working, even with complementary shaders it does not show at all. It only shows with all shaders, including internal, turned off.

Works for me at the very least with versions of OF & Complementary from this year, and older. It uses a workaround to create a similar effect. https://i.postimg.cc/D09bh5vx/glowing.png

To get the vanilla glowing you just need fast render off and internal shaders off (which has a relation to fast render).

**Comment by Ten2O** [comment]
> 
> 
> > Still not working, even with complementary shaders it does not show at all. It only shows with all shaders, including internal, turned off.
> 
> Works for me at the very least with versions of OF & Complementary from this year, and older. It uses a workaround to create a similar effect. https://i.postimg.cc/D09bh5vx/glowing.png
> 
> To get the vanilla glowing you just need fast render off and internal shaders off (which has a relation to fast render).

I have the latest versions of both and I get no glowing effect. Is there a setting I must enable? Fast render is off.

**Comment by Codewow** [comment]
It seems you need to launch vanilla at least once (and probably play on the server) for the glow effect to work.

 I'm not sure what causes this, but loading up vanilla, logging in to my server, then loading up optifine and logging in got glowing to work for me on 1.18 pre7 at least
